.jenkins/pipelines/Azure/Linux/Jenkinsfile

```diff
@@ -237,75 +237,76 @@ properties([buildDiscarder(logRotator(artifactDaysToKeepStr: '90',
 
 try{
     oe.emailJobStatus('STARTED')
+    def testing_stages = [
+        "Host verification 1604 Release":         { ACCHostVerificationTest('16.04', 'Release') },
+        "Host verification 1804 Release":         { ACCHostVerificationTest('18.04', 'Release') },
+        "ACC1804 GNU gcc SGX1FLC":                { ACCGNUTest() },
+        "RHEL-8 simulation clang-8 SGX1 Release": { RHEL8Test('clang', 'Release', ["OE_SIMULATION=1"], ["-DHAS_QUOTE_PROVIDER=OFF"]) },
```

Comment:
This is a good implementation and improvement of the current implementation of passing multiple configurations via multiple String parameters. It, however, does change the standardization from what is currently implemented and can be confusing.

I am in favor of the switch, would it be possible to alter the ACCTest configuration to be configured in the same way?

Reply:
That's a good point. I believe that we can do this.

I'll look into it.

Reply:
Apart from this, LGTM.

Reply:
I've made this change for `ACCTest` and the testing stages look cleaner now:
https://github.com/ionutbalutoiu/openenclave/blob/31ee859724c40a8a4ac9aac8cf7fc53ceda64900/.jenkins/pipelines/Azure/Linux/Jenkinsfile#L232-L269

I also updated the rest of the functions (`ACCContainerTest`, `ACCPackageTest`) from `.jenkins/pipelines/Azure/Linux/Jenkinsfile` to use the same approach: use a `List` parameter called `extra_cmake_args` instead of separate `String` parameters for each CMake flag.

The advantage of this approach is: whenever we need to add more CMake flags, we don't need to change the function implementation again.

Besides `.jenkins/pipelines/Azure/Linux/Jenkinsfile`, I also updated `.jenkins/pipelines/Agnostic/Linux/Jenkinsfile` with the same changes.

The remaining `.jenkins/pipelines/Azure/Windows/Jenkinsfile` is not updated yet, since it uses the function `WinCompilePackageTest` from the shared library that has the same old interface (multiple `String` parameters for the CMake flags).

If the above changes look good to you, we can merge this. 

I can follow up with a separate PR to update `.jenkins/pipelines/Azure/Windows/Jenkinsfile` as well, after changing the shared library method first (which is also used in the `.jenkins/pipelines/Azure/Nightly/Packaging.Jenkinsfile`). I want to be careful to not break master branch while doing it.

Reply:
Waiting for a successful E2E right now.